web/packages/teleterm/src/mainProcess/clusterLifecycleManager/clusterLifecycleManager.ts

```diff
@@ -0,0 +1,227 @@
+/**
+ * Teleport
+ * Copyright (C) 2025 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Cluster } from 'gen-proto-ts/teleport/lib/teleterm/v1/cluster_pb';
+
+import Logger from 'teleterm/logger';
+import { AwaitableSender } from 'teleterm/mainProcess/awaitableSender';
+import { RendererIpc } from 'teleterm/mainProcess/types';
+import type { WindowsManager } from 'teleterm/mainProcess/windowsManager';
+import { AppUpdater } from 'teleterm/services/appUpdater';
+import { isTshdRpcError, TshdClient } from 'teleterm/services/tshd';
+import { mergeClusterProfileWithDetails } from 'teleterm/services/tshd/cluster';
+import { RootClusterUri } from 'teleterm/ui/uri';
+
+import { ClusterStore } from '../clusterStore';
+import { ProfileChangeSet } from '../profileWatcher';
+
+export interface ClusterLifecycleEvent {
+  uri: RootClusterUri;
+  op: 'did-add-cluster' | 'will-logout' | 'will-logout-and-remove';
+}
+
+export interface ProfileWatcherError {
+  error: unknown;
+  reason: 'processing-error' | 'exited';
+}
+
+export class ClusterLifecycleManager {
```

Comment:
The class could use a JSDoc explaining the overall architecture and how `ClusterLifecycleManager` sits between the renderer, the cluster store and the profile watcher.